experimental/dds/tree2/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -80,17 +83,40 @@ export const makeNodeEditGenerator = (
 	};
 	type EditState = FuzzTestState & TreeContext;
 
-	async function setPayloadGenerator(state: EditState): Promise<FuzzSetPayload> {
+	async function setPayloadGenerator(state: EditState): Promise<FuzzNodeEditChange> {
 		const trees = state.trees;
 		const tree = trees[state.treeIndex];
 		// generate edit for that specific tree
 		const path = getExistingRandomNodePosition(tree, state.random);
-		return {
+		const setPayload: FuzzSetPayload = {
 			nodeEditType: "setPayload",
 			path,
 			value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER),
 			treeIndex: state.treeIndex,
 		};
+		switch (path.parentField) {
+			case sequenceFieldKey:
+				return {
+					type: "sequence",
+					edit: setPayload,
+				};
+			case valueFieldKey:
+				return {
+					type: "value",
+					edit: setPayload,
+				};
+			case optionalFieldKey:
+				return {
+					type: "optional",
+					edit: setPayload,
+				};
+			default:
+				// default case returns a sequenceNodeEdit
```

Comment:
You can do

```ts
case sequenceFieldKey:
default:
	return {
		type: "sequence",
		edit: setPayload
	}
```

Then it becomes obvious that the default case is the same as the sequence case and the comment is unnecessary.

Reply:
moved default case under sequenceFieldKey case